### ISSUE TITLE
BrowserState stores full AssetCollection- and Tag-Object in Session

### ISSUE THREAD

**Issue by dlubitz** [neos/neos-development-collection#3539]
Media.Browsers BrowserState stores the AssetCollection and Tag as entities in the session. On shutdown the browser state gets serialized including his data. On __sleep both entities get loaded fully, to get them serialized.

This leads to thousands of DB queries to fetch all tags and assets, to serialize them and put them into session storage. Independend of there usage. Which is kind of inperforment eager fetching at the end of the request.

The performance is very poor on media libraries with lot of assets and tags.

TOP 5 queries by count on opening media browser with ~1.700 assets and ~170 tags, just showing 30 assets per page.
```
1355x
SELECT t1.persistence_object_identifier AS persistence_object_identifier_2, t1.lastmodified AS lastmodified_3, t1.title AS title_4, t1.caption AS caption_5, t1.copyrightnotice AS copyrightnotice_6, t1.assetsourceidentifier AS assetsourceidentifier_7, t0.name AS name_8, t0.presetidentifier AS
 presetidentifier_9, t0.presetvariantname AS presetvariantname_10, t0.width AS width_11, t0.height AS height_12, t1.resource AS resource_13, t0.originalasset AS originalasset_14, t1.dtype FROM neos_media_domain_model_imagevariant t0 INNER JOIN neos_media_domain_model_asset t1 ON t0.persistence_object_identifier
 = t1.persistence_object_identifier WHERE t0.originalasset = ?

2363x
SELECT t0.persistence_object_identifier AS persistence_object_identifier_1, t0.position AS position_2, t0.imagevariant AS imagevariant_3, t0.dtype, t0.x AS x_4, t0.y AS y_5, t0.width AS width_6, t0.height AS height_7, t0.aspectratioasstring AS aspectratioasstring_8, t0.quality AS quality_9, 
t0.width AS width_10, t0.height AS height_11, t0.maximumwidth AS maximumwidth_12, t0.maximumheight AS maximumheight_13, t0.minimumwidth AS minimumwidth_14, t0.minimumheight AS minimumheight_15, t0.ratiomode AS ratiomode_16, t0.allowupscaling AS allowupscaling_17 FROM neos_media_domain_model_adjustment_abstracti
mageadjustment t0 WHERE t0.imagevariant = ? AND t0.dtype IN ('neos_media_adjustment_cropimageadjustment', 'neos_media_adjustment_qualityimageadjustment', 'neos_media_adjustment_resizeimageadjustment') ORDER BY t0.position ASC

3999x
SELECT t0.persistence_object_identifier AS persistence_object_identifier_1, t0.label AS label_2, t0.parent AS parent_3 FROM neos_media_domain_model_tag t0 INNER JOIN neos_media_domain_model_asset_tags_join ON t0.persistence_object_identifier = neos_media_domain_model_asset_tags_join.media_ta
g WHERE neos_media_domain_model_asset_tags_join.media_asset = ? ORDER BY t0.label DESC

3999x
SELECT t0.persistence_object_identifier AS persistence_object_identifier_1, t0.staticresource AS staticresource_2, t0.configuration AS configuration_3, t0.configurationhash AS configurationhash_4, t0.width AS width_5, t0.height AS height_6, t0.quality AS quality_7, t0.originalasset AS origin
alasset_8, t0.resource AS resource_9 FROM neos_media_domain_model_thumbnail t0 WHERE t0.originalasset = ?

3999x
SELECT t0.persistence_object_identifier AS persistence_object_identifier_1, t0.title AS title_2 FROM neos_media_domain_model_assetcollection t0 INNER JOIN neos_media_domain_model_assetcollection_assets_join ON t0.persistence_object_identifier = neos_media_domain_model_assetcollection_assets_
join.media_assetcollection WHERE neos_media_domain_model_assetcollection_assets_join.media_asset = ? ORDER BY t0.title ASC
```
## Possible solution
1. Just store a scalar value in BrowserState
2. Adjust serialization process and prevent loading of unloaded/lazy relations

### Steps to Reproduce

Tested in Neos 5.3.6/Flow 6.3.14 and Neos 7.1.3/Flow 7.1.5 with a clean database.

Prepare:
* Enable SQLLogger (to see what happens)
* Create asset collection "Test" in MediaBrowser
* Select created asset collection "Test"
* Upload 100 random images (in this collection)

Issue:
* Open MediaBrowser and click "All" collection
* Clear Query-Log
* Click asset collection "Test"
* Check Query-Log 
(e.g. `grep DEBUG Data/Logs/Query_Development.log | awk '{ print substr($0, 61) }' | sort | uniq -c | sort`)

It creates (TOP 8 queries by count):

***Depending on images per page***
30x Queries `neos_media_domain_model_thumbnail`
30x Queries `neos_media_domain_model_thumbnail` // another query
30x Queries `neos_media_domain_model_asset`
60x Queries `neos_flow_resourcemanagement_persistentresource`

***Depending on overall images uploaded***
100x Queries `neos_media_domain_model_tag`
100x Queries `neos_media_domain_model_thumbnail`
100x Queries `neos_media_domain_model_assetcollection`
100x Queries `neos_media_domain_model_imagevariant`

#### Expected behavior

Just expected Assets and Tags should be queried from database

#### Actual behavior

All Assets and Tags are queried from database.

### Affected Versions

Neos: 5.3. - 7.3 ( with following Flow-Versions)
Flow: 7.3.0, 7.2.2, 7.1.5, 7.0.11, 6.3.14

**Comment by dlubitz** [comment]
I could nail it down to a specific change, that causes this issue:
https://github.com/neos/flow-development-collection/commit/bd64efe8d45a49c0c4a1c0aa0fcbfe6a14412d4d

I guess this has prevented the loading of the objects before. Question is, if this is still a valid fix and we need to change the serialization within the regarding implementation (e.g medias BrowserState). Or if this change needs to be reverted as it has unexpected side effects.

**Comment by bwaidelich** [comment]
Thanks for reporting, this seems to be a major issue rendering the media module of large instances almost unusable

**Comment by bwaidelich** [comment]
@dlubitz I tried to reproduce bug and fix locally (with a Neos 7.3 instance).
For this I created 100 Tags and enabled the query log.
When opening the media module I see 31 queries containing `neos_media_domain_model_tag` with and without the patch applied.. Can you clarify the steps to reproduce a little?

**Comment by dlubitz** [comment]
@bwaidelich I updated the issue description as help for reproducing the issue.

**Comment by kdambekalns** [comment]
Another regression related to neos/flow-development-collection#2517?

**Comment by dlubitz** [comment]
Will be fixed with neos/flow-development-collection#2661

**Comment by dlubitz** [comment]
Fix has been released
